### PR TITLE
Apply phx-no-feedback permanently, unless cleared

### LINKED ISSUE #3071: Some containers do not receive the `phx-no-feedback` class even though they have the `phx-feedback-for` properly defined
### Environment

* Elixir version (elixir -v): 1.14.4
* Phoenix version (mix deps): 1.7.10
* Phoenix LiveView version (mix deps): 0.20.1 `b2c8999aaccf07509bf2f119f7a23d14269bfa26` vs `bed2ff05bb024b2927fb9c1523e42ca55627715f` "Speed up patching 5-10x (#2845)"
* Operating system: Linux Mint
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome, FF
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual behavior

Some elements (containers in `phx-feedback-for` parlance) no longer receive the `phx-no-feedback` class when they should i.e. when the `phx-feedback-for` is present with the proper form field name.

I've tried hard to detect the pattern i.e. to find in what way these elements are defined differently than those that still receive the class but so far without success - meaning that at this moment I can't write a demo app replicating this, that is for as long as I can't tell what I am supposed to replicate. 

So far I've tried the following (both isolated and in combo with others) and no combo managed to circumvent the bug :
- removing the element id
- render element unconditionally
- make the element class a static string (as opposed to composed dynamically)
- make the `phx-feedback-for` unconditional static string (as opposed conditioned by the assigns)

I've noticed the bug first when migrated from 0.20.1 to 0.20.3 and then dived into finding the exact commit. If starting from checking out `b2c8999aaccf07509bf2f119f7a23d14269bfa26` which is the last commit without the bug , the bug will not manifest until the "Bump build" in `5a3368da026ea011c2f6790bdd7045ce216402ec` but when going in reverse it will be there in  `bed2ff05bb024b2927fb9c1523e42ca55627715f ` where the relevant changes to `phoenix_live_view.js` were first made.

### Expected behavior

All elements with the `phx-feedback-for` attribute properly set should receive the `phx-no-feedback` class.

### PR DESCRIPTION
**Pull request by SteffenDE**
`phx-no-feedback` is a class that is added to elements with the `phx-feedback-for` attribute when the corresponding elements were not interacted with yet. The problem with the current approach is that LV of course removes the added class when the DOM is patched. Therefore we were relying on tracking all input elements on each patch to properly hide the feedback. This worked fine until the magicId optimization. Now, whole sections of the DOM could be skipped on a patch, not tracking inputs inside them and therefore not hiding feedback on the corresponding containers.

This all was built before we had the `JS` module, which gives us a nice way to permanently apply classes to an element, even if it is patched. Using this functionality, we can now apply the `phx-no-feedback` class correctly when we first render the page. This way, we don't need remove the class on each patch, and we don't rely on tracking all inputs in dom_patch. When we then interact with the input, or submit the form, we show the feedback again as usual, removing the sticky class from the feedback containers.

Fixes #3071.

### CODE REVIEW

**Comment by SteffenDE** [comment]
```elixir
Application.put_env(:sample, Example.Endpoint,
  http: [ip: {127, 0, 0, 1}, port: 5001],
  server: true,
  live_view: [signing_salt: "aaaaaaaa"],
  secret_key_base: String.duplicate("a", 64)
)

Mix.install([
  {:plug_cowboy, "~> 2.5"},
  {:jason, "~> 1.0"},
  {:phoenix, "~> 1.7.0"},
  {:phoenix_live_view, "0.20.4", override: true}
])

defmodule Example.ErrorView do
  def render(template, _), do: Phoenix.Controller.status_message_from_template(template)
end

defmodule Example.HomeLive do
  use Phoenix.LiveView, layout: {__MODULE__, :live}

  def mount(_params, _session, socket) do
    {:ok, assign(socket, count: 0, submit_count: 0, validate_count: 0)}
  end

  defp phx_vsn, do: Application.spec(:phoenix, :vsn)
  defp lv_vsn, do: Application.spec(:phoenix_live_view, :vsn)

  def render("live.html", assigns) do
    ~H"""
    <script src={"https://cdn.jsdelivr.net/npm/phoenix@#{phx_vsn()}/priv/static/phoenix.min.js"}></script>
    <script src={"https://cdn.jsdelivr.net/gh/SteffenDE/phoenix_live_view@apply_feedback_permanently_assets/priv/static/phoenix_live_view.js"}></script>
    <%!-- <script src="http://localhost:8000/priv/static/phoenix_live_view.js"></script> --%>
    <script src="https://cdn.tailwindcss.com"></script>
    <script>
      let liveSocket = new window.LiveView.LiveSocket("/live", window.Phoenix.Socket)
      liveSocket.connect()
    </script>
    <style>
      * { font-size: 1.1em; }

      .phx-no-feedback {
        display: none;
      }
    </style>
    <%= @inner_content %>
    """
  end

  def render(assigns) do
    ~H"""
    <p>Button Count: <%= @count %></p>
    <p>Validate Count: <%= @validate_count %></p>
    <p>Submit Count: <%= @submit_count %></p>
    <button phx-click="inc" class="bg-blue-500 text-white p-4">+</button>
    <button phx-click="dec" class="bg-blue-500 text-white p-4">-</button>

    <.myform />

    <div phx-feedback-for="myfeedback">
      I am visible, because phx-no-feedback is not set for myfeedback!
    </div>
    """
  end

  defp myform(assigns) do
    ~H"""
    <form id="myform" name="test" phx-change="validate" phx-submit="submit">
      <input type="text" name="name" class="border border-gray-500" placeholder="type sth">

      <.other_input />

      <button type="submit">Submit</button>
      <button type="reset">Reset</button>
    </form>
    """
  end

  defp other_input(assigns) do
    ~H"""
    <input type="text" name="myfeedback" class="border border-gray-500" placeholder="myfeedback">
    """
  end

  def handle_event("validate", _params, socket) do
    {:noreply, assign(socket, :validate_count, socket.assigns.validate_count + 1)}
  end

  def handle_event("submit", _params, socket) do
    {:noreply, assign(socket, :submit_count, socket.assigns.submit_count + 1)}
  end

  def handle_event("inc", _params, socket) do
    {:noreply, assign(socket, :count, socket.assigns.count + 1)}
  end

  def handle_event("dec", _params, socket) do
    {:noreply, assign(socket, :count, socket.assigns.count - 1)}
  end
end

defmodule Example.Router do
  use Phoenix.Router
  import Phoenix.LiveView.Router

  pipeline :browser do
    plug(:accepts, ["html"])
  end

  scope "/", Example do
    pipe_through(:browser)

    live("/", HomeLive, :index)
  end
end

defmodule Example.Endpoint do
  use Phoenix.Endpoint, otp_app: :sample
  socket("/live", Phoenix.LiveView.Socket)
  plug(Example.Router)
end

{:ok, _} = Supervisor.start_link([Example.Endpoint], strategy: :one_for_one)
Process.sleep(:infinity)
```

File: assets/js/phoenix_live_view/dom.js [hunk 1]
```diff
@@ -331,28 +333,40 @@ let DOM = {
...
+      // we only remove them once the elements are touched or the form is submitted,
+      // see showError
+      DOM.all(container, selector, el => JS.addOrRemoveClasses(el, [PHX_NO_FEEDBACK_CLASS], []))
     }
   },
 
-  resetForm(form, phxFeedbackFor){
+  resetForm(form, phxFeedbackFor, phxFeedbackGroup){
     Array.from(form.elements).forEach(input => {
       let query = `[${phxFeedbackFor}="${input.id}"],
```
Comment (josevalim) [inline assets/js/phoenix_live_view/dom.js:346]: Btw, wasn't the plan to stop querying it based on the ID? Perhaps now is a good time to remove it?

Comment (josevalim) [reply assets/js/phoenix_live_view/dom.js:346]: The changelog says we removed this feature on v0.19.0 :)

**Comment by josevalim** [comment]
@SteffenDE should this PR also remove the code from patching that tracks inputs?

File: assets/js/phoenix_live_view/dom.js (hunk 1 above)
Comment (SteffenDE) [reply assets/js/phoenix_live_view/dom.js]: removed

**Comment by SteffenDE** [comment]
@josevalim I refactored dompatch to track feedback containers instead of input elements
https://github.com/phoenixframework/phoenix_live_view/pull/3084/commits/4634ae58cde4bdb781d574becadb99fa7f5ab5b0

**Comment by SteffenDE** [comment]
We maybe want to use #3086 instead.

**Comment by chrismccord** [comment]
Rolling with this one for now. Will revisit #3086 when all the `input_changed?` releases are ready. Thanks! ❤️❤️❤️🐥🔥

**Pull request merged**